### ISSUE TITLE
Single-quoted variables not working for SQL highlighting

### ISSUE THREAD

**Issue by H3dg3** [PrismJS/prism#1509]
According to the examples, the following should work, but it does not
`SET @'quoted-variable' = 3; `
All other quoted variable examples do work, i.e. the following:
```
SET @variable = 1;
SET @$_ = 2;
SET @"quoted-variable" = 3;
SET @`quoted-variable` = 3;
```

**Comment by H3dg3** [comment]
Here you go: https://jsfiddle.net/zL9hbdo4/2/
I hope a fiddle is OK.

**Comment by RunDevelopment** [comment]
Thank you for providing this example.
I have found the problem and will provide a fix soon.

**Comment by RunDevelopment** [comment]
While the simple example of
```sql
SET @'quoted-variable' = 3;
SET @'quoted-variable' = 3;
```

is now working correctly, due to a greedy matching bug (#1492) the other example is still not highlighted correctly.

```sql
CREATE USER 'user'@'server-ip' IDENTIFIED BY 'password';
GRANT ALL PRIVILEGES ON scheme.* TO 'user'@'server-ip' WITH GRANT OPTION;
```

![image](https://user-images.githubusercontent.com/20878432/43516864-19f526b4-9587-11e8-804f-459b49299760.png)

**Comment by RunDevelopment** [comment]
I used a littly trick to by pass the greedy bug. I won't work 100% of the time but it should solve your issue.

![image](https://user-images.githubusercontent.com/20878432/43517457-09679546-9589-11e8-87c5-fea4d66edd5e.png)

**Comment by H3dg3** [comment]
I just tested with the last commit on my end and it works like a charm. Hope the fix makes it into the 1.16.0 :+1:
